test/DGmethods/compressible_Navier_Stokes/ex_001_dycoms-imex.jl

```diff
@@ -0,0 +1,466 @@
+end
+
+function global_max_scalar(A, mpicomm)
+  MPI.Allreduce(A, MPI.MAX, mpicomm)[1]
+end
+
+
+function extract_state(dg, localQ, ijk, e)
+    bl = dg.balancelaw
+    FT = eltype(localQ)
+    nstate = num_state(bl, FT)
+    l_Q = MArray{Tuple{nstate},FT}(undef)
+    for s in 1:nstate
+        l_Q[s] = localQ[ijk,s,e]
+    end
+    return Vars{vars_state(bl, FT)}(l_Q)
+end
+
+
+"""
+  Initial Condition for DYCOMS_RF01 LES
+@article{doi:10.1175/MWR2930.1,
+author = {Stevens, Bjorn and Moeng, Chin-Hoh and Ackerman,
+          Andrew S. and Bretherton, Christopher S. and Chlond,
+          Andreas and de Roode, Stephan and Edwards, James and Golaz,
+          Jean-Christophe and Jiang, Hongli and Khairoutdinov,
+          Marat and Kirkpatrick, Michael P. and Lewellen, David C. and Lock, Adrian and
+          Maeller, Frank and Stevens, David E. and Whelan, Eoin and Zhu, Ping},
+title = {Evaluation of Large-Eddy Simulations via Observations of Nocturnal Marine Stratocumulus},
+journal = {Monthly Weather Review},
+volume = {133},
+number = {6},
+pages = {1443-1462},
+year = {2005},
+doi = {10.1175/MWR2930.1},
+URL = {https://doi.org/10.1175/MWR2930.1},
+eprint = {https://doi.org/10.1175/MWR2930.1}
+}
+"""
+function Initialise_DYCOMS!(state::Vars, aux::Vars, (x,y,z), t)
+  FT            = eltype(state)
+  xvert::FT     = z
+  Rd::FT        = R_d
+  Rv::FT        = R_v
+  Rm::FT        = Rd
+  ϵdv::FT       = Rv/Rd
+  cpd::FT       = cp_d
+
+  # These constants are those used by Stevens et al. (2005)
+  qref::FT      = FT(9.0e-3)
+  q_tot_sfc::FT = qref
+  q_pt_sfc      = PhasePartition(q_tot_sfc)
+  Rm_sfc::FT    = gas_constant_air(q_pt_sfc)
+  T_sfc::FT     = 290.4
+  P_sfc::FT     = MSLP
+  ρ_sfc::FT     = P_sfc / Rm_sfc / T_sfc
+  # Specify moisture profiles
+  q_liq::FT      = 0
+  q_ice::FT      = 0
+  q_c::FT        = 0
+  zb::FT         = 600         # initial cloud bottom
+  zi::FT         = 840         # initial cloud top
+  ziplus::FT     = 875
+  dz_cloud       = zi - zb
+  θ_liq::FT      = 289
+  if xvert <= zi
+    θ_liq = FT(289.0)
+    q_tot = qref
+  else
+    θ_liq = FT(297.0) + (xvert - zi)^(FT(1/3))
+    q_tot = FT(1.5e-3)
+  end
+  q_c = q_liq + q_ice
+
+  ugeo = FT(7)
+  vgeo = FT(-5.5)
+  u, v, w = ugeo, vgeo, FT(0)
+
+  # --------------------------------------------------
+  # perturb initial state to break the symmetry and
+  # trigger turbulent convection
+    # --------------------------------------------------
+    randnum1   = rand(Uniform(-0.02,0.02))
+    randnum2   = rand(Uniform(-0.000015,0.000015))
+    randnum3   = rand(Uniform(-0.1,0.1))
+    randnum4   = rand(Uniform(-0.1,0.1))
+    if xvert <= 400.0
+        θ_liq += randnum1 * θ_liq
+        q_tot += randnum2 * q_tot
+        u     += randnum3 * u
+        v     += randnum4 * v
+    end
+  # --------------------------------------------------
+  # END perturb initial state
+  # --------------------------------------------------
```

Comment:
Since we are merging things and need to have reproducible results for testing at every push, I would comment out these lines that depend on a random number. 
Let's discuss it. I know we want to keep as few changes as possible.